medusa/restore_cluster.py

```diff
@@ -254,11 +246,19 @@ def _chunk(my_list, nb_chunks):
                 # associate one restore host with several backups as we don't have the same number of nodes.
                 self.host_map[restore_hosts[i]] = {'source': grouped_backups[i], 'seed': False}
 
+    def _tokenmap_to_sorted_nodes(self, tokenmap):
+        nodes = dict()
+        for node in tokenmap.keys():
+            nodes[node] = tokenmap[node]['tokens'][0]
+        return sorted(nodes.items(), key=operator.itemgetter(1))
+
     @staticmethod
     def _is_restore_in_place(backup_tokenmap, target_tokenmap):
         # If at least one node is part of both tokenmaps, then we're restoring in place
         # Otherwise we're restoring a remote cluster
-        return len(set(backup_tokenmap.keys()) & set(target_tokenmap.keys())) > 0
+        logging.info(f"backup tokenmap keys: {backup_tokenmap.keys()}")
+        logging.info(f"target tokenmap keys: {target_tokenmap.keys()}")
+        return len(list(set(backup_tokenmap.keys()) & set(target_tokenmap.keys()))) > 0
```

Comment:
Using the `&` operand allows us to get the entries that exist in both sets. We consider that a restore is an in place one if at least one common node exists in both clusters (some nodes could've been replaced since the backup was taken).
So the expression returns true if the list of overlapping nodes between the two sets is non empty.
I'm not totally sure I need to wrap this in a `list()` though.

Reply:
So we need an intersection of set(A) and set(B) ?

Didn't Python have set(A).intersection(set(B)) ? Not sure if you need to cast B even to a set actually. So len of that.

Reply:
It seems `set.intersection(set(A), set(B))` would work also as syntax.